### ISSUE TITLE
Can't import _tifffile.c in Python 3.6

### ISSUE THREAD

**Issue by AetherUnbound** [blink1073/tifffile#29]
I've run into problems attempting to install tifffile into a new conda environment running Python 3.6. It seems to import correctly in Python 2.7. 

```
mattb@AI-Linux-mattb:~/git/aicsimage$ conda create --name tifffiletest python=2 && activate tifffiletest
Fetching package metadata .........
Solving package specifications: .

Package plan for installation in environment /home/mattb/programs/anaconda2/envs/tifffiletest:

The following NEW packages will be INSTALLED:

    openssl:    1.0.2l-0     
    pip:        9.0.1-py27_1 
    python:     2.7.13-0     
    readline:   6.2-2        
    setuptools: 27.2.0-py27_0
    sqlite:     3.13.0-0     
    tk:         8.5.18-0     
    wheel:      0.29.0-py27_0
    zlib:       1.2.8-3      

Proceed ([y]/n)? y

#
# To activate this environment, use:
# > source activate tifffiletest
#
# To deactivate this environment, use:
# > source deactivate tifffiletest
#

(tifffiletest) mattb@AI-Linux-mattb:~/git/aicsimage$ pip install tifffile
Collecting tifffile
Installing collected packages: tifffile
Successfully installed tifffile-0.12.1
(tifffiletest) mattb@AI-Linux-mattb:~/git/aicsimage$ pip install numpy
Collecting numpy
  Downloading https://artifactory.corp.alleninstitute.org/artifactory/api/pypi/pypi-virtual/packages/5f/d2/9fa0201944933afd6d059f1e32aa6bdb203b23ab62fc823d3adf36295b9a/numpy-1.13.1-cp27-cp27mu-manylinux1_x86_64.whl (16.6MB)
    100% |████████████████████████████████| 16.6MB 94kB/s 
Installing collected packages: numpy
Successfully installed numpy-1.13.1
(tifffiletest) mattb@AI-Linux-mattb:~/git/aicsimage$ python
Python 2.7.13 |Continuum Analytics, Inc.| (default, Dec 20 2016, 23:09:15) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
Anaconda is brought to you by Continuum Analytics.
Please check out: http://continuum.io/thanks and https://anaconda.org
>>> import tifffile
>>> 
```

This process is fine in Python 2, but in python 3:
```
mattb@AI-Linux-mattb:~/git/aicsimage$ conda create --name tft python=3 && activate tft
Fetching package metadata .........
Solving package specifications: .

Package plan for installation in environment /home/mattb/programs/anaconda2/envs/tft:

The following NEW packages will be INSTALLED:

    openssl:    1.0.2l-0     
    pip:        9.0.1-py36_1 
    python:     3.6.1-2      
    readline:   6.2-2        
    setuptools: 27.2.0-py36_0
    sqlite:     3.13.0-0     
    tk:         8.5.18-0     
    wheel:      0.29.0-py36_0
    xz:         5.2.2-1      
    zlib:       1.2.8-3      

Proceed ([y]/n)? y

#
# To activate this environment, use:
# > source activate tft
#
# To deactivate this environment, use:
# > source deactivate tft
#

(tft) mattb@AI-Linux-mattb:~/git/aicsimage$ pip install numpy tifffile
Collecting numpy
  Downloading https://artifactory.corp.alleninstitute.org/artifactory/api/pypi/pypi-virtual/packages/59/e2/57c1a6af4ff0ac095dd68b12bf07771813dbf401faf1b97f5fc0cb963647/numpy-1.13.1-cp36-cp36m-manylinux1_x86_64.whl (17.0MB)
    100% |████████████████████████████████| 17.0MB 103kB/s 
Collecting tifffile
Installing collected packages: numpy, tifffile
Successfully installed numpy-1.13.1 tifffile-0.12.1
```
Here's where the problem occurs:
```
(tft) mattb@AI-Linux-mattb:~/git/aicsimage$ python
Python 3.6.1 |Continuum Analytics, Inc.| (default, May 11 2017, 13:09:58) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import tifffile
/home/mattb/programs/anaconda2/envs/tft/lib/python3.6/site-packages/tifffile/tifffile.py:306: UserWarning: ImportError: No module named '_tifffile'. Loading of some compressed images will be very slow. Tifffile.c can be obtained at http://www.lfd.uci.edu/~gohlke/
  "ImportError: No module named '_tifffile'. "
>>> 
```
However, if I install with anaconda I don't receive any errors:
```
(tft) mattb@AI-Linux-mattb:~/git/aicsimage$ conda install tifffile -c conda-forge
Fetching package metadata ...........
Solving package specifications: .

Package plan for installation in environment /home/mattb/programs/anaconda2/envs/tft:

The following NEW packages will be INSTALLED:

    mkl:      2017.0.3-0                    
    numpy:    1.13.1-py36_0                 
    tifffile: 0.12.1-np113py36_0 conda-forge

Proceed ([y]/n)? y

numpy-1.13.1-p 100% |########################################################################| Time: 0:00:00  34.36 MB/s
(tft) mattb@AI-Linux-mattb:~/git/aicsimage$ python
Python 3.6.1 |Continuum Analytics, Inc.| (default, May 11 2017, 13:09:58) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import tifffile
>>> 
(tft) mattb@AI-Linux-mattb:~/git/aicsimage$ 
```
Is there any way to have this properly import in Python 3 without relying on anaconda? Any thoughts?

**Comment by blink1073** [comment]
We need numpy to be already fully installed in order to build the C extension.  Can you please try installing numpy completely (via conda or pip), before installing tifffile?

**Comment by AetherUnbound** [comment]
Thanks for getting back to me. Here's my attempt to install numpy first:
```
mattb@AI-Linux-mattb:~$ conda create --name tft python=3
Fetching package metadata .........
Solving package specifications: .

Package plan for installation in environment /home/mattb/programs/anaconda2/envs/tft:

The following NEW packages will be INSTALLED:

    openssl:    1.0.2l-0     
    pip:        9.0.1-py36_1 
    python:     3.6.1-2      
    readline:   6.2-2        
    setuptools: 27.2.0-py36_0
    sqlite:     3.13.0-0     
    tk:         8.5.18-0     
    wheel:      0.29.0-py36_0
    xz:         5.2.2-1      
    zlib:       1.2.8-3      

Proceed ([y]/n)? y

#
# To activate this environment, use:
# > source activate tft
#
# To deactivate this environment, use:
# > source deactivate tft
#

mattb@AI-Linux-mattb:~$ activate tft
(tft) mattb@AI-Linux-mattb:~$ pip install numpy
Collecting numpy
  Downloading https://artifactory.corp.alleninstitute.org/artifactory/api/pypi/pypi-virtual/packages/59/e2/57c1a6af4ff0ac095dd68b12bf07771813dbf401faf1b97f5fc0cb963647/numpy-1.13.1-cp36-cp36m-manylinux1_x86_64.whl (17.0MB)
    100% |████████████████████████████████| 17.0MB 111kB/s 
Installing collected packages: numpy
Successfully installed numpy-1.13.1
(tft) mattb@AI-Linux-mattb:~$ pip install tifffile
Collecting tifffile
Installing collected packages: tifffile
Successfully installed tifffile-0.12.1
(tft) mattb@AI-Linux-mattb:~$ python -c 'import tifffile'
/home/mattb/programs/anaconda2/envs/tft/lib/python3.6/site-packages/tifffile/tifffile.py:306: UserWarning: ImportError: No module named '_tifffile'. Loading of some compressed images will be very slow. Tifffile.c can be obtained at http://www.lfd.uci.edu/~gohlke/
  "ImportError: No module named '_tifffile'. "
(tft) mattb@AI-Linux-mattb:~$ 
```
Is there another way I should attempt this?

**Comment by blink1073** [comment]
Can you please try with `pip install tifffile -v` to see if any warnings are printed?

**Comment by AetherUnbound** [comment]
TL;DR: I must have had a corrupted wheel, I removed my cached version and it installed fine (even in a fresh env). Thanks for the help and quick responses! 

```
(tft) mattb@AI-Linux-mattb:~$ pip install tifffile -v
Collecting tifffile
  ...
  Getting page https://artifactory.corp.alleninstitute.org/artifactory/api/pypi/pypi-snapshot-local/simple/tifffile/
  Looking up "https://artifactory.corp.alleninstitute.org/artifactory/api/pypi/pypi-snapshot-local/simple/tifffile/" in the cache
  No cache entry available
  "GET /artifactory/api/pypi/pypi-snapshot-local/simple/tifffile/ HTTP/1.1" 404 None
  Status code 404 not in [200, 203, 300, 301]
  Could not fetch URL https://artifactory.corp.alleninstitute.org/artifactory/api/pypi/pypi-snapshot-local/simple/tifffile/: 404 Client Error: Not Found for url: https://artifactory.corp.alleninstitute.org/artifactory/api/pypi/pypi-snapshot-local/simple/tifffile/ - skipping
  Using version 0.12.1 (newest of versions: 0.1.0, 0.2.0, 0.3.0, 0.3.1, 0.3.2, 0.3.3, 0.4, 0.5, 0.6.0, 0.6.1, 0.6.2, 0.7.0, 0.8.0, 0.9.0, 0.9.1, 0.9.2, 0.10.0, 0.11.0, 0.11.1, 0.12.0, 0.12.1)
  Using cached wheel link: file:///home/mattb/.cache/pip/wheels/41/5e/04/c6a7abcee9d12338f18172f5e933d0c7f6b4099e4c3d78cbf8/tifffile-0.12.1-cp36-none-any.whl
Installing collected packages: tifffile

Successfully installed tifffile-0.12.1
Cleaning up...
(tft) mattb@AI-Linux-mattb:~$ pip uninstall tifffile
Uninstalling tifffile-0.12.1:
  ...
  Successfully uninstalled tifffile-0.12.1
(tft) mattb@AI-Linux-mattb:~$ rm .cache/pip/wheels/41/5e/04/c6a7abcee9d12338f18172f5e933d0c7f6b4099e4c3d78cbf8/*
(tft) mattb@AI-Linux-mattb:~$ pip install tifffile -v
Collecting tifffile
  ...
  "GET /artifactory/api/pypi/pypi-snapshot-local/simple/tifffile/ HTTP/1.1" 404 None
  Status code 404 not in [200, 203, 300, 301]
  Could not fetch URL https://artifactory.corp.alleninstitute.org/artifactory/api/pypi/pypi-snapshot-local/simple/tifffile/: 404 Client Error: Not Found for url: https://artifactory.corp.alleninstitute.org/artifactory/api/pypi/pypi-snapshot-local/simple/tifffile/ - skipping
  Using version 0.12.1 (newest of versions: 0.1.0, 0.2.0, 0.3.0, 0.3.1, 0.3.2, 0.3.3, 0.4, 0.5, 0.6.0, 0.6.1, 0.6.2, 0.7.0, 0.8.0, 0.9.0, 0.9.1, 0.9.2, 0.10.0, 0.11.0, 0.11.1, 0.12.0, 0.12.1)
  Looking up "https://artifactory.corp.alleninstitute.org/artifactory/api/pypi/pypi-virtual/packages/41/d2/086654909995176e34228ed0e5673fed505651083f0d8b277867fb4d9f3a/tifffile-0.12.1.tar.gz" in the cache
  Current age based on date: 340346
  "GET /artifactory/api/pypi/pypi-virtual/packages/41/d2/086654909995176e34228ed0e5673fed505651083f0d8b277867fb4d9f3a/tifffile-0.12.1.tar.gz HTTP/1.1" 200 74149
  Downloading https://artifactory.corp.alleninstitute.org/artifactory/api/pypi/pypi-virtual/packages/41/d2/086654909995176e34228ed0e5673fed505651083f0d8b277867fb4d9f3a/tifffile-0.12.1.tar.gz (74kB)
  Downloading from URL https://artifactory.corp.alleninstitute.org/artifactory/api/pypi/pypi-virtual/packages/41/d2/086654909995176e34228ed0e5673fed505651083f0d8b277867fb4d9f3a/tifffile-0.12.1.tar.gz#md5=8a8afa74dd0df7915ac376a6cd7eeffc (from https://artifactory.corp.alleninstitute.org/artifactory/api/pypi/pypi-virtual/simple/tifffile/)
    96% |███████████████████████████████ | 71kB 28.6MB/s eta 0:00:01  Updating cache with response from "https://artifactory.corp.alleninstitute.org/artifactory/api/pypi/pypi-virtual/packages/41/d2/086654909995176e34228ed0e5673fed505651083f0d8b277867fb4d9f3a/tifffile-0.12.1.tar.gz"
  Caching due to etag
    100% |████████████████████████████████| 81kB 6.5MB/s 
  Running setup.py (path:/tmp/pip-build-osatardh/tifffile/setup.py) egg_info for package tifffile
    Running command python setup.py egg_info
    running egg_info
    creating pip-egg-info/tifffile.egg-info
    writing pip-egg-info/tifffile.egg-info/PKG-INFO
    writing dependency_links to pip-egg-info/tifffile.egg-info/dependency_links.txt
    writing top-level names to pip-egg-info/tifffile.egg-info/top_level.txt
    writing manifest file 'pip-egg-info/tifffile.egg-info/SOURCES.txt'
    warning: manifest_maker: standard file '-c' not found

    reading manifest file 'pip-egg-info/tifffile.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    warning: no files found matching '*.c'
    warning: no previously-included files matching '__pycache__' found under directory '*'
    warning: no previously-included files matching '*.py[co]' found under directory '*'
    writing manifest file 'pip-egg-info/tifffile.egg-info/SOURCES.txt'
  Source in /tmp/pip-build-osatardh/tifffile has version 0.12.1, which satisfies requirement tifffile from https://artifactory.corp.alleninstitute.org/artifactory/api/pypi/pypi-virtual/packages/41/d2/086654909995176e34228ed0e5673fed505651083f0d8b277867fb4d9f3a/tifffile-0.12.1.tar.gz#md5=8a8afa74dd0df7915ac376a6cd7eeffc
Building wheels for collected packages: tifffile
  Running setup.py bdist_wheel for tifffile ...   Destination directory: /tmp/tmpkplz10vopip-wheel-
  Running command /home/mattb/programs/anaconda2/envs/tft/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-osatardh/tifffile/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpkplz10vopip-wheel- --python-tag cp36
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.6
  creating build/lib.linux-x86_64-3.6/tifffile
  copying tifffile/__init__.py -> build/lib.linux-x86_64-3.6/tifffile
  copying tifffile/tifffile.py -> build/lib.linux-x86_64-3.6/tifffile
  running egg_info
  writing tifffile.egg-info/PKG-INFO
  writing dependency_links to tifffile.egg-info/dependency_links.txt
  writing top-level names to tifffile.egg-info/top_level.txt
  warning: manifest_maker: standard file '-c' not found

  reading manifest file 'tifffile.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  warning: no files found matching '*.c'
  warning: no previously-included files matching '__pycache__' found under directory '*'
  warning: no previously-included files matching '*.py[co]' found under directory '*'
  writing manifest file 'tifffile.egg-info/SOURCES.txt'
  copying tifffile/_tifffile.c -> build/lib.linux-x86_64-3.6/tifffile
  running build_ext
  building 'tifffile._tifffile' extension
  creating build/temp.linux-x86_64-3.6
  creating build/temp.linux-x86_64-3.6/tifffile
  gcc -pthread -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/mattb/programs/anaconda2/envs/tft/lib/python3.6/site-packages/numpy/core/include -I/home/mattb/programs/anaconda2/envs/tft/include/python3.6m -c tifffile/_tifffile.c -o build/temp.linux-x86_64-3.6/tifffile/_tifffile.o
  gcc -pthread -shared -L/home/mattb/programs/anaconda2/envs/tft/lib -Wl,-rpath=/home/mattb/programs/anaconda2/envs/tft/lib,--no-as-needed build/temp.linux-x86_64-3.6/tifffile/_tifffile.o -L/home/mattb/programs/anaconda2/envs/tft/lib -lpython3.6m -o build/lib.linux-x86_64-3.6/tifffile/_tifffile.cpython-36m-x86_64-linux-gnu.so
  installing to build/bdist.linux-x86_64/wheel
  running install
  running install_lib
  creating build/bdist.linux-x86_64
  creating build/bdist.linux-x86_64/wheel
  creating build/bdist.linux-x86_64/wheel/tifffile
  copying build/lib.linux-x86_64-3.6/tifffile/_tifffile.cpython-36m-x86_64-linux-gnu.so -> build/bdist.linux-x86_64/wheel/tifffile
  copying build/lib.linux-x86_64-3.6/tifffile/__init__.py -> build/bdist.linux-x86_64/wheel/tifffile
  copying build/lib.linux-x86_64-3.6/tifffile/_tifffile.c -> build/bdist.linux-x86_64/wheel/tifffile
  copying build/lib.linux-x86_64-3.6/tifffile/tifffile.py -> build/bdist.linux-x86_64/wheel/tifffile
  running install_egg_info
  Copying tifffile.egg-info to build/bdist.linux-x86_64/wheel/tifffile-0.12.1-py3.6.egg-info
  running install_scripts
  creating build/bdist.linux-x86_64/wheel/tifffile-0.12.1.dist-info/WHEEL
done
  Stored in directory: /home/mattb/.cache/pip/wheels/41/5e/04/c6a7abcee9d12338f18172f5e933d0c7f6b4099e4c3d78cbf8
  Removing source in /tmp/pip-build-osatardh/tifffile
Successfully built tifffile
Installing collected packages: tifffile

Successfully installed tifffile-0.12.1
Cleaning up...
(tft) mattb@AI-Linux-mattb:~$ python -c 'import tifffile'
(tft) mattb@AI-Linux-mattb:~$ 
```
Again I tried it in a new env and it picked up the new (uncorrupted?) cached wheel and installed fine.
